src/worker/bpf.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    cmp,
+    ffi::{c_char, CString},
+    fmt::Display,
+    mem, slice, thread, time,
+};
+
+use core_affinity::CoreId;
+use libc::{SYS_bpf, SYS_perf_event_open};
+use log::info;
+
+use aya_obj::copy_instructions;
+use aya_obj::generated::{
+    bpf_attach_type, bpf_attr, bpf_cmd, bpf_prog_type, perf_event_attr,
+    perf_event_sample_format, perf_type_id,
+};
+
+use crate::{BaseConfig, Worker, WorkerError, Workload, WorkloadConfig};
+
+#[derive(Debug, Clone, Copy)]
+pub struct BpfWorker {
+    config: BaseConfig,
+    workload: WorkloadConfig,
+}
+
+impl BpfWorker {
+    pub fn new(workload: WorkloadConfig, cpu: CoreId, process: usize) -> Self {
+        BpfWorker {
+            config: BaseConfig { cpu, process },
+            workload,
+        }
+    }
+}
+
+impl Worker for BpfWorker {
+    fn run_payload(&self) -> Result<(), WorkerError> {
+        info!("{self}");
+
+        let Workload::Bpf { nprogs, tracepoint } = self.workload.workload
+        else {
+            unreachable!()
+        };
+
+        for i in 0..nprogs {
+            // Prepare a bpf program attributes
+            let mut attr = unsafe { mem::zeroed::<bpf_attr>() };
+            let u = unsafe { &mut attr.__bindgen_anon_3 };
+            let prog_fd;
+
+            let mut name: [c_char; 16] = [0; 16];
+            let idx = i.to_string();
+            let cstring = CString::new(format!("berserker{idx}")).unwrap();
```

Comment:
Couldn't you just trust `format` for turning `i` to string?
```suggestion
            let cstring = CString::new(format!("berserker{i}")).unwrap();
```